src/harness/testcases/heartbeat_testcase.py

```diff
@@ -742,6 +744,68 @@ def test_WINNF_FT_S_HBT_15(self):
         datetime.strptime(response['transmitExpireTime'], '%Y-%m-%dT%H:%M:%SZ'),
         transmit_expire_time_1)
 
+  def heartbeatRequest(self, cbsd_id, grant_id):
+    # Heartbeat Request Helper to check Heartbeat Response
+    # Request Heartbeat
+    request = {
+      'heartbeatRequest': [{
+        'cbsdId': cbsd_id,
+        'grantId': grant_id,
+        'operationState': 'GRANTED'
+      }]
+    }
+
+    response = self._sas.Heartbeat(request)['heartbeatResponse'][0]
+    # Check the heartbeat response
+    self.assertEqual(response['cbsdId'], cbsd_id)
+    # Response Should fail with Code 103 or 500
+    self.assertTrue(response['response']['responseCode'] in (103, 500))
+    del request, response
+
+  @winnforum_testcase
+  def test_WINNF_FT_S_HBT_17(self):
+    """Heartbeat Request from CBSD in Registered state (immediately after CBSD's grant is expired)
+    Response Code should  be 103 or 500"""
+
+    # Register the device
+    device_a = json.load(
+      open(os.path.join('testcases', 'testdata', 'device_e.json')))
+    self._sas_admin.InjectFccId({'fccId': device_a['fccId']})
+    request = {'registrationRequest': [device_a]}
+    response = self._sas.Registration(request)['registrationResponse'][0]
+
+    # Check registration response
+    self.assertEqual(response['response']['responseCode'], 0)
+    cbsd_id = response['cbsdId']
+    del request, response
+
+    # Request grant
+    grant_0 = json.load(open(os.path.join('testcases', 'testdata', 'grant_0.json')))
+    grant_0['cbsdId'] = cbsd_id
+    request = {'grantRequest': [grant_0]}
+
+    # Check grant response
+    response = self._sas.Grant(request)['grantResponse'][0]
+    self.assertEqual(response['cbsdId'], cbsd_id)
+    self.assertEqual(response['response']['responseCode'], 0)
+    grant_id = response['grantId']
+
+    # Calculate the Difference Between Current Time and the GrantExpireTime
+    grant_expire_time = time.strptime(response['grantExpireTime'], '%Y-%m-%dT%H:%M:%SZ')
+    current_time = time.strptime(datetime.utcnow().strftime('%Y-%m-%dT%H:%M:%SZ'), '%Y-%m-%dT%H:%M:%SZ')
+    difference_time = time.mktime(grant_expire_time) - time.mktime(current_time)
+    del request, response
+
+    # Log the Wait Time in seconds
+    print "Difference between grantExpireTime and CurrentTime (in seconds)", difference_time
+
+    # Ensure that GrantExpireTime is Greater than CurrentTime
+    self.assertGreaterEqual(difference_time, 0)
+    # Start the Timer to request the Heartbeat after the calculated seconds delay from the current time
+    Timer(difference_time, lambda: self.heartbeatRequest(cbsd_id, grant_id)).start()
```

Comment:
Isn't it simpler to just do:

```
time.sleep(grant_expire_time - current_time + 1)
# send heartbeat request and check response
```

I don't think setting up a Timer is necessary and I'm worried that it might cause problems.